omod/src/test/java/org/openmrs/module/fhir2/narrative/DefaultNarrativeTemplateTest.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.narrative;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.notNullValue;
+
+import java.util.Date;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.AllergyIntolerance;
+import org.hl7.fhir.r4.model.Base64BinaryType;
+import org.hl7.fhir.r4.model.BooleanType;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Dosage;
+import org.hl7.fhir.r4.model.Duration;
+import org.hl7.fhir.r4.model.Identifier;
+import org.hl7.fhir.r4.model.Medication;
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.hl7.fhir.r4.model.Narrative;
+import org.hl7.fhir.r4.model.Period;
+import org.hl7.fhir.r4.model.Provenance;
+import org.hl7.fhir.r4.model.Quantity;
+import org.hl7.fhir.r4.model.Range;
+import org.hl7.fhir.r4.model.Reference;
+import org.hl7.fhir.r4.model.Signature;
+import org.hl7.fhir.r4.model.StringType;
+import org.hl7.fhir.r4.model.Timing;
+import org.hl7.fhir.r4.model.codesystems.AllergyintoleranceClinical;
+import org.hl7.fhir.r4.model.codesystems.AllergyintoleranceVerification;
+import org.junit.Test;
+
+public class DefaultNarrativeTemplateTest {
+	
+	@Test
+	public void testForAllergyIntolerance() {
```

Comment:
Always prefer the `should` style of naming tests over `test` to make it clear what the test is testing. For example, I would name this `shouldGenerateAnAllergyIntoleranceNarrative`.

Reply:
Rather than populating this resource completely in code like this, it seems like it would be simpler to download the file from [here](https://www.hl7.org/fhir/allergyintolerance-example.json), remove the "text" section and save that in the test resources, then load it.

For example:

```java
// ensure ctx is a FhirContext with the correct NarrativeGenerator
IParser parser = ctx.newJsonParser();

// load initial data (this is our "given" for the test)
AllergyIntolerance given = parser.parseResource(AllergyIntolerance.class, getClass().getClassLoader().getResourceAsStream("allergyintolerance-example.json"));

AllergyIntolerance result = parser.parseResource(AllergyIntolerance.class, parser.encodeResourceToString(allergyIntolerance));

// now our assertions
assertThat(result, notNullValue());
assertThat(result.getText(), notNullValue());
assertThat(result.getText().getStatusAsString(), equalTo("generated"));
// note that we should check the generated result against a reference of what we expect it to be
assertThat(result.getText().getDivAsString(), equalTo(getClass().getClassLoader().getResourceAsStream("expected/allergyintolerance-example.html"));
```

Reply:
@ibacher Thanks a lot for this! I myself was looking for alternative ways.